### PR TITLE
Fix #5029: Invalid window limit can cause crash

### LINKED ISSUE #5029: Invalid window_limit setting can cause crash
**OS:** Windows 10
**Version:** 0.0.6
**Commit/Build:** 7c7b67f

Setting window_limit to 0(or can parsed to 0) by editing config.ini will cause immediate crash.
Setting window_limit to below 4(like 1) will cause crash when trying to open any window(new game, config window, etc)

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

### PR DESCRIPTION
**Pull request by wolfreak99**
I also fix a small bug I made when implementing this, which was assuming OpenRCT2's clamp function was the same as many other programs functions.

### CODE REVIEW

File: src/openrct2/interface/window.c [hunk 1]
```diff
@@ -403,6 +403,16 @@ void window_set_window_limit(sint32 value)
...
 {
+	//Number must always be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
+	}
+	if (gConfigGeneral.window_limit > WINDOW_LIMIT_MAX) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MAX;
+		config_save_default();
+	}
```
Comment (IntelOrca) [inline src/openrct2/interface/window.c]: Why would you want to do this every time a window is created?

Comment (Gymnasiast) [inline src/openrct2/interface/window.c]: This would better be moved to the initialisation state, where it loads the config.

Comment (wolfreak99) [inline src/openrct2/interface/window.c]: @IntelOrca @Gymnasiast You guys make a point.. a question though, I havent explored just yet, but does the config file ever get loaded again after the opening initialisation? window_set_window_limit handles the clamping when it's set in game, but not outside the game.

Comment (IntelOrca) [inline src/openrct2/interface/window.c]: > but does the config file ever get loaded again after the opening initialisation

no

**Comment by Gymnasiast** [comment]
@IntelOrca Could you merge this if it's up to your standard now?

**Comment by IntelOrca** [comment]
I still don't think this is the right place for it. How about `config_open`?

**Comment by wolfreak99** [comment]
i'm off my meds and half asleep, but later we should consider maybe having something like "config_validate" that can be used to ensure config settings are set properly. just an idea.

File: src/openrct2/config.c [hunk 2]
```diff
@@ -544,6 +544,17 @@ bool config_open(const utf8string path)
 
 	free(lineBuffer);
 	SDL_RWclose(file);
+
+	// Window limit must be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
```
Comment (IntelOrca) [inline src/openrct2/config.c]: You don't need this anymore

File: src/openrct2/config.c [hunk 3]
```diff
@@ -544,6 +544,17 @@ bool config_open(const utf8string path)
...
+	}
+	else if (gConfigGeneral.window_limit > WINDOW_LIMIT_MAX) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MAX;
+		config_save_default();
```
Comment (IntelOrca) [inline src/openrct2/config.c]: You don't need this anymore

File: src/openrct2/config.c (hunk 2 above)
Comment (wolfreak99) [inline src/openrct2/config.c]: why would we not want it to update the config file? wouldn't being notified that the game didn't accept a specific value be useful? like, if i set a value to 3, and the game doesn't accept 3, i wouldn't know for sure if my file was still saying 3 yet it was showing 4 windows. however if it updated it to 4 ande i seen it next time i go to modify the file, i couldn think "oh so whatever i just typed wasnt accepted"

Comment (IntelOrca) [inline src/openrct2/config.c]: I think you are overthinking this. I think it should just be removed and then this PR can be merged.

Another reason this is wrong is that this is a function for just loading a config file, nothing else - and not necessarily the default config file. So you could end up loading a completely different config file and it then saves to the default config file.

**Pull request merged**